src/kv/kv.h

```diff
@@ -1495,10 +1523,12 @@ namespace kv
         e->rollback(v);
     }
 
-    DeserialiseSuccess deserialise(
+    DeserialiseSuccess deserialise_views(
       const std::vector<uint8_t>& data,
       bool public_only = false,
-      Term* term = nullptr) override
+      bool commit = true,
```

Comment:
I don't think you need the `commit` bool here. Instead, you can just check that if `tx == nullptr`, then we should commit, `else` we should call `tx->set_view_list(views)`.

Reply:
I agree, I was trying to leave it open for the case where you might pass in a transaction but also want to commit but it is best not to add code that you don't need yet.